src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs

```diff
@@ -86,7 +86,56 @@ public IAzureContextContainer DefaultProfile
             set
             {
                 _profile = value;
+                var accountMap = CreateAccountMap(AzureRmProfileProvider.Instance.Profile.Accounts);
+                foreach (var currentAccount in _profile.Accounts)
+                {
+                    if (accountMap.ContainsKey(currentAccount.Id))
```

Comment:
```suggestion
                foreach (var currentAccount in _profile.Accounts)
                {
                    if (accountMap.ContainsKey(currentAccount.Id))
```
You can use `public bool TryGetValue (TKey key, out TValue value);`